### PR TITLE
Support theming `CupertinoSwitch`s

### PR DESCRIPTION
**Pull request by guidezpl**
## Description

`CupertinoSwitch`s currently don't use the ambient `CupertinoThemeData`. This means switches created with `Switch.adaptive` (https://github.com/flutter/flutter/issues/94821) always use the cupertino default green.

In order to support theming them in a non-breaking manner, this PR introduces:
- `applyTheme` flag for `CupertinoSwitch` and `Switch.adaptive`
- `CupertinoThemeData.applyThemeToAll`, which serves as a default value. It defaults to `false`. This name was chosen to be generic enough to support future use cases. In the future, it could be also be made `true` in a breaking change and made subject to removal
- a code sample showing theming opt-in and single-switch opt-out

## Related issues
- https://github.com/flutter/flutter/issues/94821
- Unblocks b/236234608

## Tests
- Added a golden test for `CupertinoSwitch` that can respect theming opt-in and single-switch opt-out
- Updated existing `CupertinoThemeData` tests
- Fixed existing code sample test

### CODE REVIEW

File: examples/api/lib/material/switch/switch.3.dart [hunk 1]
```diff
@@ -0,0 +1,68 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// Flutter code sample for [Switch].
```
Comment (esouthren) [inline examples/api/lib/material/switch/switch.3.dart:5]: Should this doc specify that this is a Cupertino-specific example?

**Review by esouthren — APPROVED** [review]
LGTM!

**Comment by CaseyHillers** [comment]
@guidezpl the Google testing check appears to be stuck as your PR's branch is ~40 commits behind tip of tree. Can you try rebasing to get it unstuck?

**Comment by guidezpl** [comment]
Looks like it unstuck itself

File: examples/api/lib/material/switch/switch.3.dart (hunk 1 above)
Comment (guidezpl) [reply examples/api/lib/material/switch/switch.3.dart:5]: This isn't super important, the description that's used in docs does mention it

**Pull request merged**